### PR TITLE
gh-76961: Possible fix for buildbot failures in test_pep3118

### PR DESCRIPTION
**Pull request by mdickinson**
This PR is an attempt to fix the buildbot failures introduced by my merge of #5561, by restricting the relevant tests to something that should work on both 32-bit and 64-bit platforms.

* Issue: gh-76961

### CODE REVIEW

**Review by eric-wieser — APPROVED** [review]
Thanks! If this fails too then I guess we should just only run the tests of PEP3118 when we know the actual memory layout (according to the ctypes module) is what we expected, and otherwise skip then

**Comment by mdickinson** [comment]
Argh. Windows wants `'T{<b:x:3x<L:y:}'` rather than `'T{<b:x:3x<I:y:}'`.

**Comment by mdickinson** [comment]
Added a couple of commits to silence warnings.

File: Modules/_ctypes/stgdict.c [hunk 1]
```diff
@@ -355,8 +355,8 @@ _ctypes_alloc_format_padding(const char *prefix, Py_ssize_t padding)
         return _ctypes_alloc_format_string(prefix, "x");
     }
 
-    int ret = PyOS_snprintf(buf, sizeof(buf), "%zdx", padding);
-    assert(0 <= ret && ret < sizeof(buf));
+    int ret = PyOS_snprintf(buf, sizeof(buf), "%zdx", padding); (void)ret;
```
Comment (mdickinson) [inline Modules/_ctypes/stgdict.c:358]: FWIW, the `(void)ret;` trick comes from the LLVM Coding Standard: https://llvm.org/docs/CodingStandards.html#assert-liberally

**Comment by mdickinson** [comment]
One buildbot failure, for buildbot/aarch64 RHEL8 Refleaks PR, which looks unrelated (`test_importlib` and `test_asyncio` failures).

@eric-wieser This seems to be working; do you have time to give this a second-pass review?

**Review by eric-wieser — APPROVED** [review]
Thanks, LGTM

**Pull request merged**